numpy/core/tests/test_umath.py

```diff
@@ -686,6 +686,58 @@ def test_float_remainder_corner_cases(self):
                 assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, rem))
 
 
+class TestDivisionOverflows:
+    overflow_results = {
+        np.remainder: '0',
+        np.fmod: '0',
+        np.floor_divide: '-np.iinfo(dividend_dtype).min',
+        np.divmod: '(-np.iinfo(dividend_dtype).min, 0)'
+    }
+
+    @pytest.mark.parametrize("dividend_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("divisor_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("operation",
+            [np.remainder, np.fmod, np.divmod, np.floor_divide])
+    def test_overflows_no_raises(self, dividend_dtype, divisor_dtype, operation):
+        arrays = [np.array(
+                    [dividend_dtype(
+                        np.iinfo(dividend_dtype).min)]*i
+                    ) for i in range(1, 129)]
+        divisor = np.array(-1, dtype=divisor_dtype)
+        with np.errstate(divide='raise'):
+            # If dividend is a larger type than the divisor (`else` case),
+            # then, result will be a larger type than dividend and will not
+            # result in an overflow for `divmod` and `floor_divide`.
+            if np.dtype(dividend_dtype).itemsize >= np.dtype(
```

Comment:
@seberg , the tests are right yeah? I can't figure out why the arrays are not getting upcasted like the scalars, any pointers here will be helpful.